aws/ecs/client.go

```diff
@@ -0,0 +1,68 @@
+package ecs
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	awsConfig "github.com/aws/aws-sdk-go-v2/config"
+	awsECS "github.com/aws/aws-sdk-go-v2/service/ecs"
+)
+
+type Client struct {
+	awsECS.Client
+
+	ctx context.Context
+}
+
+func NewECSClient(ctx context.Context, region string) (*Client, error) {
+	cfg, err := awsConfig.LoadDefaultConfig(ctx,
+		awsConfig.WithRegion(region),
+	)
+	if err != nil {
+		return nil, err
+	}
+	return &Client{Client: *awsECS.NewFromConfig(cfg), ctx: ctx}, nil
```

Comment:
If `NewFromConfig` returns `*awsECS.Client` then it would make sense  that `Client` has the same type ( `*awsECS.Client` vs  `awsECS.Client`).

Reply:
Thank you, I thought struct embedding required values vs reference